lib/event_store_client/client.rb

```diff
@@ -5,9 +5,12 @@
 module EventStoreClient
   class Client
     NoCallMethodOnSubscriber = Class.new(StandardError)
+    WrongExpectedEventVersion = Class.new(StandardError)
 
     def publish(stream:, events:, expected_version: nil)
```

Comment:
We need a retry mechanism here of there was a conflict. I suggest to have up to 5 retries here